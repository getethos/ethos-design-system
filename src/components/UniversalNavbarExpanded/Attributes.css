.body {
  --HoverOpacity: 0.78;
}
