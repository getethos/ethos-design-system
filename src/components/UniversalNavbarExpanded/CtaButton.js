import React from 'react'
import PropTypes from 'prop-types'

// EDS core components
import { Button } from '../index'

// Styles
import styles from './CtaButton.module.scss'

/**
 * Simple CTA button for use in UniversalNavBarExpanded.
 *
 * TODO Add onKeyPress for tracking function
 *
 * @param {string} href - URL for the button to link to
 * @param {function} trackingFunction - Analytics function run when CTA Button is clicked
 * @param {boolean} hideOnMobile - Hide the CTA on phone only
 * @param {string} copy - Copy text for the button
 *
 * @return {JSX.Element}
 */
const CtaButton = ({ href, trackingFunction, hideOnMobile, copy }) => {
  // We still rely on some legacy UniversalNavbar styles from FancyAnimatedLogo.scss
  // TODO convert these to module.scss capable styles
  const CtaButtonClasses = [
    'cta-button',
    'navbar-expanded',
    'show-when-scrolled',
  ]

  if (hideOnMobile) {
    CtaButtonClasses.push(styles.hidden)
  }

  return (
    <a
      className={CtaButtonClasses.join(' ')}
      onClick={trackingFunction}
      href={href}
    >
      <Button.Small.Black>{copy}</Button.Small.Black>
    </a>
  )
}

CtaButton.propTypes = {
  href: PropTypes.string.isRequired,
  trackingFunction: PropTypes.func.isRequired,
  hideOnMobile: PropTypes.bool,
  copy: PropTypes.string.isRequired,
}

export default CtaButton
