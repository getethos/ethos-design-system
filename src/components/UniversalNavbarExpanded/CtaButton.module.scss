@import '../Media/Media.scss';
@import '../Variables.scss';

@include for-phone-only {
  .hidden {
    display: none;
  }
}

.visible {
  z-index: $z-index-navbar;
}
