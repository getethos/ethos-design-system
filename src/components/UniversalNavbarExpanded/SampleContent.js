/**
 * Sample of UniversalNavbarExpanded links.
 * For demonstration in styleguide & test files (.md/.test.js files) only!
 * Real implementation of this structure should use uuid() for id values!
 *
 * TODO when Kustomer implementation is ready and we want to include a talk button
 *      add TALK to the object with href & title
 */
export const CMS_LINKS = {
  // These are used e.g. in the logo, icons and bottom of mobile nav:
  INDEX: { href: '/' },
  CTA: {
    href: '/term',
    title: 'Check my price',
  },
  ACCOUNT: {
    href: '/login/',
    title: 'Account',
  },
  SEARCH: {
    href: '/search/',
    title: 'Search',
  },
  // These are the main top level link and their subnav items
  NAVLINKS: [
    {
      id: 'NAVLINKS_MOCK_ID_1',
      title: 'What we offer',
      subnav: {
        cta: {
          href: '/#/Components/UniversalNavbarExpanded',
          title: 'Our plans',
          subcopy:
            'Learn more about term life insurance and the plans and options you have available.',
          id: 'NAVLINKS_MOCK_ID_2',
        },
        items: [
          {
            id: 'NAVLINKS_MOCK_ID_3',
            href: '/#/Components/UniversalNavbarExpanded',
            title: 'What is term life insurance and how much does it cost?',
          },
          {
            id: 'NAVLINKS_MOCK_ID_4',
            href: '/how-it-works/',
            title: 'How our application process works',
          },
          {
            id: 'NAVLINKS_MOCK_ID_5',
            href: '/life/who-needs-life-insurance/',
            title: 'Do I need life insurance?',
          },
          {
            id: 'NAVLINKS_MOCK_ID_6',
            href: '/life/how-choose-right-type-life-insurance/',
            title: 'Choosing your coverage amount and term length',
          },
          {
            id: 'NAVLINKS_MOCK_ID_7',
            href: '/app/needs/',
            title: 'Coverage calculator',
          },
        ],
      },
    },
    {
      id: 'NAVLINKS_MOCK_ID_8',
      title: 'Life insurance basics',
      subnav: {
        cta: {
          href: '/life/life-insurance-basics/',
          title: 'Life insurance 101',
          subcopy:
            'Wondering where to start? We’ve broken down the essentials for you.',
          id: 'NAVLINKS_MOCK_ID_9',
        },
        items: [
          {
            id: 'NAVLINKS_MOCK_ID_10',
            href: '/life/term-group-whole/',
            title:
              'What are the main types of life insurance and how do they work?',
          },
          {
            id: 'NAVLINKS_MOCK_ID_11',
            href: '/life/underwriting-explained/',
            title: 'What is underwriting and how long does it take?',
          },
          {
            id: 'NAVLINKS_MOCK_ID_12',
            href: '/life/pick-beneficiary/',
            title: 'Choosing your beneficiary',
          },
          {
            id: 'NAVLINKS_MOCK_ID_13',
            href: '/life/who-needs-sp/',
            title: "What if I don't work?",
          },
          {
            id: 'NAVLINKS_MOCK_ID_14',
            href: '/life/employer-sponsored-life-insurance/',
            title: 'Is my life insurance through work enough?',
          },
        ],
      },
    },
    {
      id: 'NAVLINKS_MOCK_ID_15',
      title: 'Why Ethos',
      subnav: {
        cta: {
          href: '/why-ethos/',
          title: 'The Ethos Difference',
          subcopy:
            'We put people before profit. Find out how we bring our policyholders the best experience possible.',
          id: 'NAVLINKS_MOCK_ID_16',
        },
        items: [
          {
            id: 'NAVLINKS_MOCK_ID_17',
            href: '/about/',
            title: 'Our mission',
          },
          {
            id: 'NAVLINKS_MOCK_ID_18',
            href: '/reviews/',
            title: 'Reviews',
          },
          {
            id: 'NAVLINKS_MOCK_ID_19',
            href: '/life/customer-stories/',
            title: 'Customer stories',
          },
          {
            id: 'NAVLINKS_MOCK_ID_20',
            href: '/life/ethosforgood/',
            title: 'Ethos for Good',
          },
          {
            id: 'NAVLINKS_MOCK_ID_21',
            href: '/friends/',
            title: 'Refer a friend',
          },
        ],
      },
    },
    {
      id: 'NAVLINKS_MOCK_ID_22',
      title: 'Help Center',
      subnav: {
        cta: {
          href: '/faq/',
          title: 'FAQ',
          subcopy:
            'Life insurance can be complicated, but don’t worry. We’re here to help answer your questions.',
          id: 'NAVLINKS_MOCK_ID_23',
        },
        items: [
          {
            id: 'NAVLINKS_MOCK_ID_24',
            href: '/faq/life-insurance-basics/',
            title: 'Questions about life insurance',
          },
          {
            id: 'NAVLINKS_MOCK_ID_25',
            href: '/faq/ethos/',
            title: 'Questions about Ethos',
          },
          {
            id: 'NAVLINKS_MOCK_ID_26',
            href: '/faq/application/',
            title: 'Questions about your application',
          },
          {
            id: 'NAVLINKS_MOCK_ID_27',
            href: '/faq/policy/',
            title: 'Questions about your policy',
          },
          {
            id: 'NAVLINKS_MOCK_ID_28',
            href: '/contact-us/',
            title: 'Contact us',
          },
        ],
      },
    },
  ],
}
