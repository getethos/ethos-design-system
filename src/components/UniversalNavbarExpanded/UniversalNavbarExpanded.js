import React, { useState } from 'react'
import PropTypes from 'prop-types'
import get from 'lodash/get'

import FancyAnimatedLogo from '../UniversalNavbar/FancyAnimatedLogo'
import LogoNotAnimated from '../UniversalNavbar/assets/ethos-logo-black.js'
import LogoWhite from '../UniversalNavbar/assets/ethos-logo-white.js'
import {
  AccountIcon,
  SearchIcon,
  AccordionToggleIcon,
  DropdownParentIcon,
  DropdownLinkIcon,
} from '../UniversalNavbar/assets/icons.js'
import { Button, Layout, TitleMedium, TitleSmall, Footnote } from '../index'
import NavLink from '../UniversalNavbar/NavLink'
import TransformingBurgerButton from '../UniversalNavbar/TransformingBurgerButton/TransformingBurgerButton'
import DropDownCta from './DropDownCta'
import styles from './UniversalNavbarExpanded.module.scss'

// TODO move this to helpers?
const ENTER_KEY_CODE = 13
const isEnterKeyPress = (event) => {
  if (
    event.key === 'Enter' ||
    event.which === ENTER_KEY_CODE ||
    event.keyCode === ENTER_KEY_CODE
  ) {
    return true
  } else {
    return false
  }
}
// TODO: more spacing var usage in scss

// TODO: less nesting in scss

// TODO: replace white with color var in scss

// TODO: 504px that's an odd value

// TODO: alternate caret weight for DropdownParentIcon pre:hover

// TODO: 'talk to us' CTA integration once Kustomer is implemented

// TODO: adjust width of bottom forest border on parent links

// TODO: smallest laptop size, lower width between subitems w/shorter text

// TODO: set subnav height to largest

// TODO: arrow on same line as individual link

// TODO: replace blue outline on focus with cleaner white underline

// TODO: remove clickability of right side of accordion submenu
// TODO:  z index variables

// TODO: 3/2/20 QA: Add 1px stroke (#000000 20%) beneath default navbar and underneath dropdown when it's open (low priority)
// TODO: 3/2/20 QA: Reduce padding between "check my price" CTA and hamburger icon on scrolled mobile nav (low priority) right 44px override not working for scrolled CTA in navbar on mobile
// TODO: 3/2/20 QA: Side arrow breaks to new line for one of the menu items on laptop size (low priority)

// TODO:  Split main file into separate component files
// TODO:  Update test descriptions / more tests
// TODO:  JS Docs
// TODO:  more code commenting in general
// TODO:  PropTypes review

const UniversalNavbarExpanded = ({
  LinkComponent,
  hideMobileCta,
  hideDesktopCta,
  logoHref,
  trackCtaClick,
  links,
}) => {
  const [showMobileMenu, setShowMobileMenu] = useState(false)
  const [activeAccordionItem, setActiveAccordionItem] = useState(false)

  const toggleHamburger = () => {
    if (!showMobileMenu) {
      setShowMobileMenu(true)
    } else {
      setShowMobileMenu(false)
    }
  }

  const handleHamburgerKeyPress = (event) => {
    if (isEnterKeyPress(event)) {
      toggleHamburger()
    }
  }

  const toggleAccordionItem = (toggledItem) => {
    setActiveAccordionItem(
      activeAccordionItem === toggledItem ? false : toggledItem
    )
  }

  const handleAccordionItemKeyPress = (event, index) => {
    if (isEnterKeyPress(event)) {
      toggleAccordionItem(index)
    }
  }

  const NavLinkReduced = ({ className, key, href, children }) => (
    <NavLink
      className={className}
      key={key ? key : null}
      href={href}
      LinkComponent={LinkComponent}
    >
      {children}
    </NavLink>
  )

  NavLinkReduced.propTypes = {
    className: PropTypes.string,
    key: PropTypes.string,
    href: PropTypes.string,
    children: PropTypes.node,
  }

  const CtaButton = (showWhenScrolled) => {
    const CtaButtonClasses = ['cta-button']
    if (showWhenScrolled) {
      CtaButtonClasses.push('show-when-scrolled')
    }
    return (
      <a
        className={CtaButtonClasses.join(' ')}
        onClick={trackCtaClick}
        href={links.TERM.href}
      >
        <Button.Small.Black>Check my price</Button.Small.Black>
      </a>
    )
  }

  const SearchIconLink = () => (
    <NavLinkReduced className={styles.searchIcon} href={links.SEARCH.href}>
      <SearchIcon />
    </NavLinkReduced>
  )

  const AccountIconLink = () => (
    <NavLinkReduced className={styles.accountIcon} href={links.ACCOUNT.href}>
      <AccountIcon />
    </NavLinkReduced>
  )

  const AccordionNav = () => (
    <div className={styles.accordion}>
      {links.NAVLINKS.map((link, idx) => (
        <div
          key={link.id}
          className={
            idx === activeAccordionItem
              ? [styles.accordionItem, styles.active].join(' ')
              : styles.accordionItem
          }
          onClick={() => toggleAccordionItem(idx)}
          onKeyPress={(e) => handleAccordionItemKeyPress(e, idx)}
          tabIndex={0}
          role="button"
        >
          <div className={styles.accordionParent}>
            <TitleMedium.Sans.Regular400>
              {link.title}
            </TitleMedium.Sans.Regular400>
            <AccordionToggleIcon />
          </div>
          <div className={styles.accordionChildren}>
            <div
              key={get(link, 'subnav.cta.id')}
              className={styles.accordionChild}
            >
              <TitleSmall.Sans.Light300>
                <NavLinkReduced href={get(link, 'subnav.cta.href')}>
                  {get(link, 'subnav.cta.title')}
                </NavLinkReduced>
              </TitleSmall.Sans.Light300>
            </div>
            {get(link, 'subnav.items').map((link) => (
              <div key={link.id} className={styles.accordionChild}>
                <TitleSmall.Sans.Light300>
                  <NavLinkReduced href={link.href}>{link.title}</NavLinkReduced>
                </TitleSmall.Sans.Light300>
              </div>
            ))}
          </div>
        </div>
      ))}
    </div>
  )

  const BelowAccordion = () => (
    <div className={styles.belowAccordion}>
      <div className={styles.belowAccordionItem}>
        <NavLinkReduced href={links.TERM.href}>
          <TitleMedium.Sans.Regular400>
            {links.TERM.title}
          </TitleMedium.Sans.Regular400>
        </NavLinkReduced>
      </div>
      <div className={styles.belowAccordionItem}>
        <NavLinkReduced href={links.ACCOUNT.href}>
          <TitleMedium.Sans.Regular400>
            {links.ACCOUNT.title}
          </TitleMedium.Sans.Regular400>
        </NavLinkReduced>
      </div>
      <div className={styles.belowAccordionItem}>
        <NavLinkReduced href={links.SEARCH.href}>
          <TitleMedium.Sans.Regular400>
            {links.SEARCH.title}
          </TitleMedium.Sans.Regular400>
        </NavLinkReduced>
      </div>
    </div>
  )

  const MobileMenu = () => (
    <div
      className={
        showMobileMenu
          ? [styles.phoneAndTablet, styles.visible].join(' ')
          : styles.phoneAndTablet
      }
    >
      <div
        className={showMobileMenu ? styles.mobileMenu : styles.hideMobileMenu}
      >
        <NavLinkReduced href={logoHref} className={styles.phoneLogo}>
          {LogoWhite({ className: styles.logo })}
        </NavLinkReduced>

        <div className={styles.hamburger}>
          <TransformingBurgerButton
            showMobileMenu={showMobileMenu}
            clickHandler={() => toggleHamburger()}
            keyPressHandler={(e) => handleHamburgerKeyPress(e)}
          />
        </div>
        <AccordionNav />
        <BelowAccordion />
      </div>
      {/* Mobile menu items, getAnEstimate only shows when scrolled */}
      <NavLinkReduced href={logoHref} className={styles.phoneLogoFancy}>
        <FancyAnimatedLogo />
      </NavLinkReduced>
      {!hideMobileCta && CtaButton(true)}
      <SearchIconLink />
    </div>
  )

  const DropDownChildren = ({ child }) => (
    <div className={styles.dropdownNavChildren}>
      <Layout.HorizontallyPaddedContainer>
        <div className={styles.dropdownNavChildrenInner}>
          <div className={styles.dropdownNavChildrenCta}>
            {child.subnav &&
              get(child, 'subnav.cta') &&
              get(child, 'subnav.cta.href') && (
                <NavLinkReduced href={get(child, 'subnav.cta.href')}>
                  <DropDownCta
                    title={get(child, 'subnav.cta.title')}
                    subcopy={get(child, 'subnav.cta.subcopy')}
                  />
                </NavLinkReduced>
              )}
          </div>
          <div className={styles.dropdownNavChildrenItems}>
            {child.subnav.items.map((link) => (
              <div className={styles.dropdownNavChild} key={link.id}>
                <Footnote.Regular400>
                  <NavLinkReduced href={link.href}>
                    <>
                      <span>{link.title}</span>
                      <DropdownLinkIcon />
                    </>
                  </NavLinkReduced>
                </Footnote.Regular400>
              </div>
            ))}
          </div>
        </div>
      </Layout.HorizontallyPaddedContainer>
    </div>
  )

  DropDownChildren.propTypes = {
    child: PropTypes.object,
  }

  const DesktopMenu = () => (
    <div className={styles.laptopAndUp}>
      <div className={styles.laptopAndUpContainer}>
        {/* Desktop menu items to the left */}
        <div className={`${styles.flex} ${styles.itemsCenter}`}>
          <NavLinkReduced href={logoHref}>
            {LogoNotAnimated({ className: styles.logo })}
          </NavLinkReduced>
          <div className={styles.dropdownNav}>
            {links.NAVLINKS.map((link) => (
              <div className={styles.dropdownNavParent} key={link.id}>
                <Footnote.Regular400>
                  {link.title}
                  <span className={styles.dropdownNavParentIcon}>
                    <DropdownParentIcon />
                  </span>
                </Footnote.Regular400>
                <DropDownChildren child={link} />
              </div>
            ))}
          </div>
        </div>

        {/* Desktop menu items to the right */}
        <div className={`${styles.flex} ${styles.itemsCenter}`}>
          <SearchIconLink />
          <AccountIconLink />
          <div className={styles.cta}>
            {!hideDesktopCta && CtaButton(false)}
          </div>
        </div>
      </div>
    </div>
  )

  return (
    <div className={styles.blockNavbar}>
      <div className={styles.navbar}>
        <Layout.ScrollDetector>
          <div className={styles.hamburger}>
            <TransformingBurgerButton
              showMobileMenu={showMobileMenu}
              clickHandler={() => toggleHamburger()}
              keyPressHandler={(e) => handleHamburgerKeyPress(e)}
            />
          </div>
          <MobileMenu />
          <DesktopMenu />
        </Layout.ScrollDetector>
      </div>
    </div>
  )
}

UniversalNavbarExpanded.propTypes = {
  /** Hide cta on mobile viewport */
  hideMobileCta: PropTypes.bool,
  /** Hide cta on desktop */
  hideDesktopCta: PropTypes.bool,
  /** Run analytics function when CTA Button gets clicked */
  trackCtaClick: PropTypes.func,
  /** agnotistic Reach and React Router Link */
  LinkComponent: PropTypes.object,
  /** Href for the logo */
  logoHref: PropTypes.string.isRequired,
  /** Object of URLs and text */
  // TODO add shape
  links: PropTypes.object.isRequired,
}

UniversalNavbarExpanded.defaultProps = {
  hideMobileCta: false,
  hideDesktopCta: false,
  logoHref: '/',
  trackCtaClick: () => {},
  links: {},
}

export { UniversalNavbarExpanded }
