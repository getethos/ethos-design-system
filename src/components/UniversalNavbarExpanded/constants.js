import uuidv4 from 'uuid/v4'
export const LINKS = {
  // These are used e.g. in the logo, icons and bottom of mobile nav:
  INDEX: { href: '/' },
  TERM: {
    href: '/term',
    title: 'Check my price',
    id: uuidv4(),
  },
  // TALK: {
  //     id: uuidv4(),
  //     href: '/',
  //     title: 'Talk to us',
  //   },
  ACCOUNT: {
    href: '/login/',
    title: 'Account',
    id: uuidv4(),
  },
  SEARCH: {
    href: '/search/',
    title: 'Search',
    id: uuidv4(),
  },

  // These are the main top level link and their subnav items
  NAVLINKS: [
    {
      id: uuidv4(),
      href: null,
      title: 'What we offer',
      subnav: {
        cta: {
          href: '/insurance/term-life-insurance/',
          title: 'Our plans',
          subcopy:
            'Learn more about term life insurance and the plans and options you have available.',
          id: uuidv4(),
        },
        items: [
          {
            id: uuidv4(),
            href: '/life/ethos-term-insurance/',
            title: 'What is term life insurance and how much does it cost?',
          },
          {
            id: uuidv4(),
            href: '/how-it-works/',
            title: 'How our application process works',
          },
          {
            id: uuidv4(),
            href: '/life/who-needs-life-insurance/',
            title: 'Do I need life insurance?',
          },
          {
            id: uuidv4(),
            href: '/life/how-choose-right-type-life-insurance/',
            title: 'Choosing your coverage amount and term length',
          },
          {
            id: uuidv4(),
            href: '/app/needs/',
            title: 'Coverage calculator',
          },
        ],
      },
    },
    {
      id: uuidv4(),
      href: null,
      title: 'Life insurance basics',
      subnav: {
        cta: {
          href: '/life/life-insurance-basics/',
          title: 'Life insurance 101',
          subcopy:
            'Wondering where to start? We’ve broken down the essentials for you.',
          id: uuidv4(),
        },
        items: [
          {
            id: uuidv4(),
            href: '/life/term-group-whole/',
            title:
              'What are the main types of life insurance and how do they work?',
          },
          {
            id: uuidv4(),
            href: '/life/underwriting-explained/',
            title: 'What is underwriting and how long does it take?',
          },
          {
            id: uuidv4(),
            href: '/life/pick-beneficiary/',
            title: 'Choosing your beneficiary',
          },
          {
            id: uuidv4(),
            href: '/life/who-needs-sp/',
            title: "What if I don't work?",
          },
          {
            id: uuidv4(),
            href: '/life/employer-sponsored-life-insurance/',
            title: 'Is my life insurance through work enough?',
          },
        ],
      },
    },
    {
      id: uuidv4(),
      href: null,
      title: 'Why Ethos',
      subnav: {
        cta: {
          href: '/why-ethos/',
          title: 'The Ethos Difference',
          subcopy:
            'We put people before profit. Find out how we bring our policyholders the best experience possible.',
          id: uuidv4(),
        },
        items: [
          {
            id: uuidv4(),
            href: '/about/',
            title: 'Our mission',
          },
          {
            id: uuidv4(),
            href: '/reviews/',
            title: 'Reviews',
          },
          {
            id: uuidv4(),
            href: '/life/customer-stories/',
            title: 'Customer stories',
          },
          {
            id: uuidv4(),
            href: '/life/ethosforgood/',
            title: 'Ethos for Good',
          },
          {
            id: uuidv4(),
            href: '/friends/',
            title: 'Refer a friend',
          },
        ],
      },
    },
    {
      id: uuidv4(),
      href: null,
      title: 'Help Center',
      subnav: {
        cta: {
          href: '/faq/',
          title: 'FAQ',
          subcopy:
            'Life insurance can be complicated, but don’t worry. We’re here to help answer your questions.',
          id: uuidv4(),
        },
        items: [
          {
            id: uuidv4(),
            href: '/faq/life-insurance-basics/',
            title: 'Questions about life insurance',
          },
          {
            id: uuidv4(),
            href: '/faq/ethos/',
            title: 'Questions about Ethos',
          },
          {
            id: uuidv4(),
            href: '/faq/application/',
            title: 'Questions about your application',
          },
          {
            id: uuidv4(),
            href: '/faq/policy/',
            title: 'Questions about your policy',
          },
          {
            id: uuidv4(),
            href: '/contact-us/',
            title: 'Contact us',
          },
        ],
      },
    },
  ],
}
