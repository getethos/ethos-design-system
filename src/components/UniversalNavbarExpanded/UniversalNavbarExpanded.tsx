import * as React from 'react'
// EDS core components
import { Caption2, Layout } from '../index'
import { LinkItem, Links } from '../types'
// Reused assets from UniversalNavbar
import LogoNotAnimated from '../UniversalNavbar/assets/ethos-logo-black'
import { AccountIcon, SearchIcon } from '../UniversalNavbar/assets/icons'
// UniversalNavbarExpanded simple siblings
import CtaButton from './CtaButton'
// UniversalNavbarExpanded complex siblings
import DropdownNav from './DropdownNav/DropdownNav'
import MobileNav from './MobileNav/MobileNav'
import NavLink from './NavLink'
// Styles
import styles from './UniversalNavbarExpanded.module.scss'
type UniversalNavbarExpandedProps = {
  hideMobileCta?: boolean
  hideDesktopCta?: boolean
  trackCtaClick: (...args: any[]) => any
  LinkComponent?: React.ElementType
  logoHref: string
  links: Links
  estimateExperiment?: boolean
}
const UniversalNavbarExpanded: React.FC<UniversalNavbarExpandedProps> = ({
  LinkComponent,
  hideMobileCta,
  hideDesktopCta,
  logoHref,
  trackCtaClick,
  links,
  estimateExperiment,
}) => {
  const BELOW_ACCORDION_LINKS = [links.CTA, links.ACCOUNT, links.SEARCH].filter(
    Boolean
  ) as LinkItem[]
  const SearchIconLink = () => (
    <NavLink className={styles.searchIcon} href={links.SEARCH?.href || ''}>
      <SearchIcon />
    </NavLink>
  )
  const AccountIconLink = () => (
    <NavLink className={styles.accountIcon} href={links.ACCOUNT?.href || ''}>
      <AccountIcon />
    </NavLink>
  )
  const ExperimentCopy = () => (
    <div
      className={styles.estimateCopyOptimizely}
      data-optimizely="estimateNavBarCopyOptimizely"
    >
      <div className={styles.estimateCopy}>
        <Caption2.Regular400>Want to know your real rate?</Caption2.Regular400>
      </div>
    </div>
  )
  const layoutClasses = [styles.flex, styles.itemsCenter]
  return (
    <div className={styles.navbarWrapper}>
      <div className={styles.navbar}>
        <Layout.ScrollDetector>
          <MobileNav
            extraClass={'isFixedCta'}
            logoHref={logoHref}
            links={links}
            secondaryLinksLinks={BELOW_ACCORDION_LINKS}
            hideMobileCta={hideMobileCta}
            ctaButtonTrackingFunction={trackCtaClick}
            LinkComponent={LinkComponent}
          />
          <div className={styles.laptopAndUp}>
            <div className={styles.laptopAndUpContainer}>
              <div className={layoutClasses.join(' ')}>
                <NavLink href={logoHref} LinkComponent={LinkComponent}>
                  {LogoNotAnimated({ className: styles.logo })}
                </NavLink>
                <DropdownNav links={links} LinkComponent={LinkComponent} />
              </div>
              <div className={layoutClasses.join(' ')}>
                {estimateExperiment && <ExperimentCopy />}
                <SearchIconLink />
                <AccountIconLink />
                <div className={styles.cta}>
                  {!hideDesktopCta && (
                    <CtaButton
                      href={links.CTA?.href || ''}
                      trackingFunction={trackCtaClick}
                    />
                  )}
                </div>
              </div>
            </div>
          </div>
        </Layout.ScrollDetector>
      </div>
    </div>
  )
}
UniversalNavbarExpanded.defaultProps = {
  hideMobileCta: false,
  hideDesktopCta: false,
  logoHref: '/',
  trackCtaClick: () => {},
  links: {},
  estimateExperiment: false,
}
export { UniversalNavbarExpanded }
