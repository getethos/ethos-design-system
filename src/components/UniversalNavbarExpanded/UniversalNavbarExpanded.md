This element has a fixed position at the top of the screen which you can see here. Also, it behaves very differently on mobile.

```jsx
import { CMS_LINKS } from './constants.js';
<UniversalNavbarExpanded logoHref={'http://test.com'} links={CMS_LINKS}/>
```
