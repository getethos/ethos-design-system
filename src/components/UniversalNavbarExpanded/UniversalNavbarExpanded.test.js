import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { UniversalNavbarExpanded } from './UniversalNavbarExpanded'
import { CMS_LINKS } from './SampleContent'

describe('<UniversalNavbarExpanded>', () => {
  describe('matches snapshot', () => {
    it('builds with sample links provided', () => {
      expect(
        shallowSnapshot(<UniversalNavbarExpanded links={CMS_LINKS} />)
      ).toMatchSnapshot()
    })
  })
})

function shallowSnapshot(jsx) {
  const renderer = new ShallowRenderer()
  renderer.render(jsx)
  return renderer.getRenderOutput()
}
