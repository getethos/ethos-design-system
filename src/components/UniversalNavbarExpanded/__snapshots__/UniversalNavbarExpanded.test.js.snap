// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UniversalNavbarExpanded> matches snapshot with no arguments 1`] = `
<div
  className="blockNavbar"
>
  <div
    className="navbar"
  >
    <ScrollDetector
      element="div"
      offsetHeight={0}
    >
      <div
        className="hamburger"
      >
        <TransformingBurgerButton
          clickHandler={[Function]}
          keyPressHandler={[Function]}
          showMobileMenu={false}
        />
      </div>
      <MobileMenu />
      <DesktopMenu />
    </ScrollDetector>
  </div>
</div>
`;
